mdsuite/calculators/coordination_number_calculation.py

```diff
@@ -109,267 +140,279 @@ def __init__(self, **kwargs):
 
-        # Check that more than one peak exists. If not, the GS search cannot be
-        # performed.
-        if len(peaks) < 2:
+        # Check that the required number of peaks exist.
+        if len(peaks) < required_peaks:
             msg = (
-                "Not enough peaks were found for the minimum analysis (First shell). Try"
-                " adjusting the filter parameters or re-calculating the RDF for a"
-                " smoother function."
+                "Not enough peaks were detecting in the RDF to perform the desired "
+                "analysis. Try reducing the number of desired peaks or improving the "
+                "quality of the RDF provided."
             )
             log.error(msg)
             raise CannotPerformThisAnalysis(msg)
         else:
-            return [peaks[0], peaks[1], peaks[2]]  # return peaks if they exist
+            return peaks
 
-    def _find_minima(self):
+    def _find_minima(self, radii: np.ndarray, rdf: np.ndarray) -> dict:
         """
         Use min finding algorithm to determine the minima of the function
 
+        Parameters
+        ----------
+        radii : np.ndarray
+                A numpy array of radii values.
+        rdf : np.ndarray
+                A numpy array of rdf values.
         Returns
         -------
-        cn_indices_1, cn_indices_2 : tuple
-                Returns a tuple of indices which can be evaluated on the CN function to
-                get the correct values.
+        coordination_shells : dict
+                A dictionary of coordination shell radial ranges.
         """
 
-        peaks = self._get_max_values()  # get the max value indices
+        peaks = self._get_rdf_peaks(rdf)  # get the max value indices
 
         # Calculate the range in which the coordination numbers should exist.
-        cn_radii_1 = golden_section_search(
-            [self.radii, self.rdf], self.radii[peaks[1]], self.radii[peaks[0]]
-        )
-        cn_radii_2 = golden_section_search(
-            [self.radii, self.rdf], self.radii[peaks[2]], self.radii[peaks[1]]
-        )
-
-        # Locate the indices of the radii values
-        cn_indices_1 = list(
-            [
-                np.where(self.radii == cn_radii_1[0])[0][0],
-                np.where(self.radii == cn_radii_1[1])[0][0],
-            ]
-        )
-        cn_indices_2 = list(
-            [
-                np.where(self.radii == cn_radii_2[0])[0][0],
-                np.where(self.radii == cn_radii_2[1])[0][0],
-            ]
-        )
+        coordination_shells = {}
+        for i in range(self.args.number_of_shells):
+            coordination_shells[i] = np.zeros(2, dtype=int)
+            cn_radii_range = golden_section_search(
+                [radii, rdf], radii[peaks[i + 1]], radii[peaks[i]]
+            )
+            for j in range(2):
+                coordination_shells[i][j] = np.where(radii == cn_radii_range[j])[0][0]
```

Comment:
I guess `np.squeeze(np.where(radii==cn_radii_range[j]))` would also work - is it a single value? But also no need to change that.